fault-tolerance/fault-tolerance/src/main/java/io/helidon/faulttolerance/TimeoutImpl.java

```diff
@@ -52,11 +67,20 @@ public String name() {
 
     @Override
     public <T> T invoke(Supplier<? extends T> supplier) {
+        if (metricsEnabled) {
+            callsCounterMetric.increment();
+        }
+
+        long start = metricsEnabled ? System.nanoTime() : 0L;
         if (!currentThread) {
             try {
-                return CompletableFuture.supplyAsync(supplier, executor)
+                T res = CompletableFuture.supplyAsync(supplier, executor)
                         .orTimeout(timeoutMillis, TimeUnit.MILLISECONDS)
                         .get();
+                if (metricsEnabled) {
+                    executionDurationMetric.record(System.nanoTime() - start, TimeUnit.NANOSECONDS);
+                }
```

Comment:
IIUC this updates the timer only if the work completes successfully? Is that what we want?

Reply:
Good point, for consistency it should be on finally I think